Formula/luajit.rb

```diff
@@ -1,18 +1,20 @@
+# NOTE: We have a policy of building only from tagged commits, but make a
+#       singular exception for luajit. This exception will not be extended
+#       to other formulae. See:
+#       https://github.com/Homebrew/homebrew-core/pull/99580
 class Luajit < Formula
   desc "Just-In-Time Compiler (JIT) for the Lua programming language"
   homepage "https://luajit.org/luajit.html"
+  # Update this to the tip of the `v2.1` branch at the start of every month.
+  url "https://github.com/LuaJIT/LuaJIT/archive/4c2441c16ce3c4e312aaefecc6d40c4fe21de97c.tar.gz"
+  # Use the version scheme `2.1.0-beta3-yyyymmdd.xx` where `yyyymmdd` is the date of the
+  # latest commit at the time of updating, and `xx` is the number of commits on that date.
+  version "2.1.0-beta3-20220623.02"
+  sha256 "e46561743772f2e338e9dbdc1a6b7c8210688e9d25dd823f93890a44a8d30b16"
   license "MIT"
   head "https://luajit.org/git/luajit-2.0.git", branch: "v2.1"
 
-  stable do
-    url "https://luajit.org/download/LuaJIT-2.0.5.tar.gz"
-    sha256 "874b1f8297c697821f561f9b73b57ffd419ed8f4278c82e05b48806d30c1e979"
-
-    # Fix for macOS 10.15 Catalina SDK
-    # https://github.com/LuaJIT/LuaJIT/issues/521
-    patch :DATA
-  end
-
+  # TODO: Fix livecheck.
   livecheck do
     url "https://luajit.org/download.html"
     regex(/href=.*?LuaJIT[._-]v?(\d+(?:\.\d+)+)\.t/i)
```

Comment:
```suggestion
  livecheck do
    url "https://github.com/LuaJIT/LuaJIT/commits/v2.1"
    regex(%r{<li.+?datetime=["']?(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z).+?href=.*?/commit/\h+["' >]}im)
    strategy :page_match do |page, regex|
      newest_date = nil
      date_count = 0
      page.scan(regex).map do |match|
        date = Date.parse(match[0])
        newest_date ||= date
        break if date != newest_date

        date_count += 1
      end
      next if newest_date.blank? || date_count.zero?

      # The main LuaJIT version is rarely updated, so we recycle it from the
      # `version` to avoid having to fetch another page.
      version.to_s.sub(/\d+\.\d+$/, "#{newest_date.strftime("%Y%m%d")}.#{date_count}")
    end
```

Unless there's a better option I'm missing, it seems like we have to parse the commit information for the branch on GitHub. This suggestion is kind of gnarly but it basically matches the `datetime` attribute of `relative-time` elements (within commit list items) to identify the newest date and the number of commits on that date. This is then substituted into the current `version` string to produce the livecheck version.

It's technically possible to add another request that properly identifies the current version from the [LuaJIT download page](https://luajit.org/download.html) (i.e., `2.1.0-beta3`) but this hasn't changed since 2017, so it may not be worth the extra overhead in this particular situation. This would rely on someone manually noticing the version change and modifying the formula version (which seems reasonable considering how notable it will be if LuaJIT officially releases a new version in the future).

Reply:
Nice; I like it. Though now I wonder if we should just be appending then time of the latest commit rather than the number of commits on that day.

Would that be simpler? It's probably a bit less easy to read, but that ship might've already sailed.

Reply:
> Though now I wonder if we should just be appending then time of the latest commit rather than the number of commits on that day.
> 
> Would that be simpler? It's probably a bit less easy to read, but that ship might've already sailed.

Yep, if you're thinking of a version format like `20220623071043`, the `strategy` block is much simpler:

```suggestion
  livecheck do
    url "https://github.com/LuaJIT/LuaJIT/commits/v2.1"
    regex(/<relative-time[^>]+?datetime=["']?(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z)["' >]/im)
    strategy :page_match do |page, regex|
      page.scan(regex).map { |match| match[0].tr("-:TZ", "") }
    end
```

Reply:
Oh, that is much worse to read, though. Also, the [commit](https://github.com/LuaJIT/LuaJIT/commit/4c2441c16ce3c4e312aaefecc6d40c4fe21de97c.patch) shows a time of `7:10:09`, which seems to be slightly different from the one parsed from the `livecheck` URL, `07:10:43`.

It might be simpler to count commits after all, barring the extra complication in the `livecheck` block.

Reply:
Ok, I've added an update so that the `test` breaks whenever the `2.1.0-beta3` part of the version changes.